### PR TITLE
Add Resource Manager Tags support to 'google_container_cluster'

### PR DESCRIPTION
**Pull request by maxi-cit**
Part of: https://github.com/hashicorp/terraform-provider-google/issues/16614

Adds support for Resource Manager Tags for GKE cluster resource

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
container: added `node_config.resource_manager_tags` field to `google_container_cluster` resource
```

```release-note:enhancement
container: added `node_config.resource_manager_tags` field to `google_container_node_pool` resource
```

### CODE REVIEW

**Comment by zli82016** [comment]
Please let me know when the PR is ready for review.

File: mmv1/third_party/terraform/services/compute/resource_compute_instance.go.erb [hunk 1]
```diff
@@ -635,7 +635,7 @@ func ResourceComputeInstance() *schema.Resource {
 				},
 			},
 
-	        	"params": {
```
Comment (LucaPrete) [inline mmv1/third_party/terraform/services/compute/resource_compute_instance.go.erb:638]: It seems changes to resource_compute_instance have nothing to do with the content of the patch and should be probably addressed in a different patch

Comment (maxi-cit) [reply mmv1/third_party/terraform/services/compute/resource_compute_instance.go.erb:638]: Hello Lucas ofc, this was due to my local autoformater I guess, but I am correcting it

File: mmv1/third_party/terraform/services/container/node_config.go.erb [hunk 2]
```diff
@@ -675,6 +675,14 @@ func schemaNodeConfig() *schema.Schema {
 						},
 					},
 				},
+				<% if version == 'beta' -%>
```
[the remark this answers is not in the document]
Comment (maxi-cit) [reply mmv1/third_party/terraform/services/container/node_config.go.erb]: thanks Satish I already solved

**Comment by LucaPrete** [comment]
Hello @maxi-cit . Any chance you can share timelines on this one please?

**Comment by maxi-cit** [comment]
Hello @LucaPrete, I had a hard time during the tests but I guess I figure it out. Fingers crossed!

**Comment by maxi-cit** [comment]
Hello @zli82016 could you review it please

**Review by skmatti — APPROVED** [review]

File: mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb [hunk 3]
```diff
@@ -9442,4 +9558,65 @@ func testAccContainerCluster_withWorkloadALTSConfig(projectID, name, networkName
...
+}
+
+resource "google_tags_tag_value" "value" {
+  parent = "tagKeys/${google_tags_tag_key.key.name}"
+  short_name = "foo-%[2]s"
+  description = "For foo resources"
+}
+
+data "google_container_engine_versions" "uscentral1a" {
+	location = "us-central1-a"
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb:9593]: The indentation should be two spaces.

File: mmv1/third_party/terraform/services/container/resource_container_node_pool_test.go.erb [hunk 4]
```diff
@@ -4207,4 +4237,80 @@ resource "google_container_node_pool" "without_confidential_boot_disk" {
 }
 `, cluster, networkName, subnetworkName, np)
 }
+
+func testAccContainerNodePool_resourceManagerTags(projectID, clusterName, networkName, subnetworkName, randomSuffix string) string {
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/container/resource_container_node_pool_test.go.erb:4241]: The indentation should be two spaces in the resource configurations.

File: mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb [hunk 5]
```diff
@@ -1438,6 +1438,12 @@ func ResourceContainerCluster() *schema.Resource {
 								},
 							},
 						},
+						"resource_manager_tags": {
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb:1441]: Can you please add this new field to the resource documentation file?
mmv1/third_party/terraform/website/docs/r/container_cluster.html.markdown

File: mmv1/third_party/terraform/services/container/node_config.go.erb [hunk 6]
```diff
@@ -675,6 +675,12 @@ func schemaNodeConfig() *schema.Schema {
 						},
 					},
 				},
+				"resource_manager_tags": {
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/container/node_config.go.erb:678]: Can you please add this field to the `resource_container_node_pool` documentation file as well?

**Review by zli82016 — CHANGES_REQUESTED** [review]

**Comment by zli82016** [comment]
Can you please also fix `terraform-provider-google-build-and-unit-tests`? Thanks.

File: mmv1/third_party/terraform/services/container/node_config.go.erb [hunk 7]
```diff
@@ -675,6 +675,12 @@ func schemaNodeConfig() *schema.Schema {
...
+					Type:        schema.TypeMap,
+					Optional:    true,
+					ForceNew:    true,
+					Description: `A map of resource manager tags. Resource manager tag keys and values have the same definition as resource manager tags. Keys must be in the format tagKeys/{tag_key_id}, and values are in the format tagValues/456. The field is ignored (both PUT & PATCH) when empty.`,
```
Comment (kamalmarhubi) [inline mmv1/third_party/terraform/services/container/node_config.go.erb:682]: can this be made a bit more user friendly? eg, the command line tool allows the following formats:
- `tagKeys/123456789=tagValues/987654321`
- `12345678901/env=dev` (org number)
- `example-project/env=dev`
- `11223344556/env=dev` (project number)

([docs](https://cloud.google.com/kubernetes-engine/docs/how-to/tags-firewall-policies#tag-syntax-cli))

it appears the API accepts tags in some of those formats?

https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1/ResourceManagerTags

the change to the documentation also includes those formats https://github.com/GoogleCloudPlatform/magic-modules/pull/9531/files#diff-77af46e13814452e05453e79bc81a3fde5b5299a0f0cbb7c12f16692b4258358R908

File: mmv1/third_party/terraform/services/container/node_config.go.erb [hunk 8]
```diff
@@ -675,6 +675,12 @@ func schemaNodeConfig() *schema.Schema {
 						},
 					},
 				},
+				"resource_manager_tags": {
+					Type:        schema.TypeMap,
+					Optional:    true,
+					ForceNew:    true,
```
[the remark this answers is not in the document]
Comment (kamalmarhubi) [reply mmv1/third_party/terraform/services/container/node_config.go.erb:681]: I've only used this feature with node pools and not the node pool autoconfig, but it does support in-place update.

**Review by kamalmarhubi — COMMENTED** [review]
@maxi-cit anything folks can do to help this get merged? we have some pretty awful local-exec workarounds to manage tags on our node pools and would love this to land in the beta provider :-)

File: mmv1/third_party/terraform/services/container/resource_container_node_pool_test.go.erb [hunk 9]
```diff
@@ -4207,4 +4237,80 @@ resource "google_container_node_pool" "without_confidential_boot_disk" {
...
+	}
+  }
+
+# Separately Managed Node Pool
+resource "google_container_node_pool" "primary_nodes" {
+  name       = google_container_cluster.primary.name
+  location   = var.region
+  cluster    = google_container_cluster.primary.name
+
+  version = data.google_container_engine_versions.gke_version.release_channel_latest_version["STABLE"]
```
[the remark this answers is not in the document]
Comment (maxi-cit) [reply mmv1/third_party/terraform/services/container/resource_container_node_pool_test.go.erb]: updated

File: mmv1/third_party/terraform/services/container/node_config.go.erb (hunk 8 above)
[the remark this answers is not in the document]
Comment (maxi-cit) [reply mmv1/third_party/terraform/services/container/node_config.go.erb:681]: removed

File: mmv1/third_party/terraform/services/container/node_config.go.erb [hunk 10]
```diff
@@ -884,6 +890,10 @@ func expandNodeConfig(v interface{}) *container.NodeConfig {
 		nc.ResourceLabels = m
 	}
 
+	if v, ok := nodeConfig["resource_manager_tags"]; ok {
+		nc.ResourceManagerTags = expandResourceManagerTags(v)
```
[the remark this answers is not in the document]
Comment (maxi-cit) [reply mmv1/third_party/terraform/services/container/node_config.go.erb:894]: condition added

File: mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb (hunk 5 above)
Comment (maxi-cit) [reply mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb:1441]: added to node_pool_auto_config block

File: mmv1/third_party/terraform/services/container/resource_container_node_pool_test.go.erb (hunk 4 above)
Comment (maxi-cit) [reply mmv1/third_party/terraform/services/container/resource_container_node_pool_test.go.erb:4241]: done

File: mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb [hunk 11]
```diff
@@ -9442,4 +9558,65 @@ func testAccContainerCluster_withWorkloadALTSConfig(projectID, name, networkName
...
+	project_id = "%[1]s"
+}
+
+resource "google_project_iam_binding" "tags" {
+	project = "%[1]s"
+	role    = "roles/resourcemanager.tagHoldAdmin"
+	members = [
+		"serviceAccount:service-${data.google_project.project.number}@container-engine-robot.iam.gserviceaccount.com",
+	]
+}
```
[the remark this answers is not in the document]
Comment (maxi-cit) [reply mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb]: done

File: mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb (hunk 3 above)
Comment (maxi-cit) [reply mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb]: done

File: mmv1/third_party/terraform/services/container/node_config.go.erb (hunk 6 above)
Comment (maxi-cit) [reply mmv1/third_party/terraform/services/container/node_config.go.erb:678]: this references the resource_container_clusters node_config block so no need

File: mmv1/third_party/terraform/services/container/node_config.go.erb (hunk 7 above)
Comment (maxi-cit) [reply mmv1/third_party/terraform/services/container/node_config.go.erb:682]: Hello @kamalmarhubi, I guess the schema description (taken from API discovery) is kinda missleading, but this enhancement also accepts other formats as it is mentioned in the documention.

File: mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb [hunk 12]
```diff
@@ -9442,4 +9558,65 @@ func testAccContainerCluster_withWorkloadALTSConfig(projectID, name, networkName
...
+
+resource "google_tags_tag_key" "key" {
+  parent = "projects/%[1]s"
+  short_name = "foobarbaz-%[2]s"
+  description = "For foo/bar resources"
+  purpose = "GCE_FIREWALL"
+  purpose_data = {
+	network = "%[1]s/%[4]s"
```
[the remark this answers is not in the document]
Comment (maxi-cit) [reply mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb]: done

**Comment by sanmaym** [comment]
Hello, I'm following this feature for a customer.. Can someone give an ETA on when this can be released?

**Comment by maxi-cit** [comment]
Hello @sanmaym I am adding in-place update support. Just fixing the broken tests

**Comment by sanmaym** [comment]
> Hello @sanmaym I am adding in-place update support. Just fixing the broken tests
Thank you. I look forward to seeing this

**Comment by rileykarson** [comment]
Adding @slevenick as a reviewer as the original reviewer is OOO

**Comment by sanmaym** [comment]
@slevenick Please let us know if you can expedite this. I know you are just the reviewer, but if you can help with the PR, that would be great .. we have a large customer eagerly waiting for this feature

**Review by slevenick — CHANGES_REQUESTED** [review]
Looks like there's a compilation error causing the build to fail:

google-beta/services/container/node_config.go:986:21: invalid array length string

**Comment by sanmaym** [comment]
> Hello @sanmaym I am adding in-place update support. Just fixing the broken tests

@maxi-cit can you please take a look at the feedback from the reviewer? Please comment if you dont have the bandwidth to pick this up.

File: mmv1/third_party/terraform/services/container/resource_container_node_pool.go.erb [hunk 13]
```diff
@@ -1605,6 +1605,48 @@ func nodePoolUpdate(d *schema.ResourceData, meta interface{}, nodePoolInfo *Node
...
+			if v, ok := d.GetOk(prefix + "node_config.0.resource_manager_tags"); ok {
+				req.ResourceManagerTags = expandResourceManagerTags(v)
+			}
+
+			// sets resource manager tags to the empty list when user removes a previously defined list of tags entriely
+			// aka the node pool goes from having tags to no longer having any
+			if req.ResourceManagerTags == nil {
+				tags := make(map[string]string)
+				rmTags := &container.ResourceManagerTags{
+					Tags: tags,
```
Comment (slevenick) [inline mmv1/third_party/terraform/services/container/resource_container_node_pool.go.erb:1621]: Do we need to add ForceSendField here like we do in the other call?

File: mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb [hunk 14]
```diff
@@ -2922,16 +2959,44 @@ func TestAccContainerCluster_withAutopilotNetworkTags(t *testing.T) {
...
+	clusterName := fmt.Sprintf("tf-test-cluster-%s", randomSuffix)
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckContainerClusterDestroyProducer(t),
 		Steps: []resource.TestStep{
 			{
-				Config: testAccContainerCluster_withAutopilot(pid, containerNetName, clusterName, "us-central1", true, true, ""),
+				Config: testAccContainerCluster_withAutopilot(pid, containerNetName, clusterName, "us-central1", randomSuffix, true, false, true, ""),
```
Comment (slevenick) [inline mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb:2999]: Why do we add this to the autopilot test?

File: mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb [hunk 15]
```diff
@@ -56,6 +56,40 @@ func TestAccContainerCluster_basic(t *testing.T) {
 	})
 }
 
+func TestAccContainerCluster_resourceManagerTags(t *testing.T) {
```
Comment (slevenick) [inline mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb:59]: Can you add an update test? Where we use the functionality for update added here? Specifically we want to test out a couple operations:
1. Adding a tag
2. Removing a tag
3. Removing all tags

**Review by slevenick — CHANGES_REQUESTED** [review]

**Comment by maxi-cit** [comment]
Hello @sanmaym, almost there. got a little confused because node_pools and clusters shared the same schema. Additionally, there is no current support for in-place update for the API for clusters but for node_pool_auto_config and node_pools

File: mmv1/third_party/terraform/services/container/resource_container_node_pool.go.erb (hunk 13 above)
Comment (maxi-cit) [reply mmv1/third_party/terraform/services/container/resource_container_node_pool.go.erb:1621]: Yes, indeed. This was causing me a litte confusion

File: mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb (hunk 14 above)
Comment (maxi-cit) [reply mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb:2999]: I tried to follow the way the auto pilot tests were implemented. I guess the intention was to make it reusable

File: mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb (hunk 15 above)
Comment (maxi-cit) [reply mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb:59]: Hello @slevenick, there is not current in-place support for these operations but for node_pools and node_auto_config

Comment (slevenick) [reply mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb:59]: Then the field should have ForceNew: true I think?

**Comment by maxi-cit** [comment]
Hello @sanmaym @slevenick could you run the tests again please? I fixed the tests and added latest suggestions.

PD: how you check coverage?

File: mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb (hunk 15 above)
Comment (maxi-cit) [reply mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb:59]: there is a mid function that forces specifically for cluster resource

**Comment by maxi-cit** [comment]
> Hi there, I'm the Modular magician. I've detected the following information about your changes:
> 
> ## Diff report
> Your PR generated some diffs in downstreams - here they are.
> 
> Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-9531-old..auto-pr-9531) ( 7 files changed, 320 insertions(+), 26 deletions(-)) Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-9531-old..auto-pr-9531) ( 7 files changed, 657 insertions(+), 30 deletions(-))
> 
> ## Missing test report
> Your PR includes resource fields which are not covered by any test.
> 
> Resource: `google_container_cluster` (334 total tests) Please add an acceptance test which includes these fields. The test should include the following:
> 
> ```terraform
> resource "google_container_cluster" "primary" {
>   node_pool {
>     node_config {
>       resource_manager_tags = # value needed
>     }
>   }
>   node_pool_auto_config {
>     resource_manager_tags = # value needed
>   }
> }
> ```

Hey @sanmaym  @slevenick, what this error refers to? I can't have both fields set at the same time

**Comment by maxi-cit** [comment]
Hello @sanmaym @slevenick, could you give me a hint on why I am getting this undefined functions? I have my test passed locally

**Comment by sanmaym** [comment]
I will defer this to Sam. I'm not sure.

*Sanmay Mishra*

Infrastructure Cloud Consultant

804-319-9657 <(804)%319-9657> Mobile




*“I choose to work flexibly & send emails outside normal office hours. **Please
do not feel obligated to respond out of your normal working hours."*

**Comment by slevenick** [comment]
It's a beta vs GA issue. In the beta provider things are fine, but in the GA provider the functions are called but not defined. Is this feature supposed to go in the beta or GA provider?

File: mmv1/third_party/terraform/services/container/resource_container_node_pool_test.go.erb [hunk 16]
```diff
@@ -4207,4 +4261,277 @@ resource "google_container_node_pool" "without_confidential_boot_disk" {
...
+
+    resource_manager_tags = {
+      "tagKeys/${google_tags_tag_key.key1.name}" = "tagValues/${google_tags_tag_value.value1.name}"
+    }
+  }
+}
+`, projectID, randomSuffix, clusterName, networkName, subnetworkName)
+}
+
+func testAccContainerNodePool_resourceManagerTagsUpdate1(projectID, clusterName, networkName, subnetworkName, randomSuffix string) string {
```
Comment (slevenick) [inline mmv1/third_party/terraform/services/container/resource_container_node_pool_test.go.erb:4357]: This whole block is within a <unless version.... tag, so it only shows up in the beta provider.

**Comment by sanmaym** [comment]
I think this (
https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1/ResourceManagerTags)
is available in GA

*Sanmay Mishra*

Infrastructure Cloud Consultant

804-319-9657 <(804)%319-9657> Mobile




*“I choose to work flexibly & send emails outside normal office hours. **Please
do not feel obligated to respond out of your normal working hours."*

**Comment by maxi-cit** [comment]
Hello @sanmaym @slevenick I fixed the annotations issue but I guess the diff report will fail. any hint on this? the API does not allow to have them both at the same time. The behavior I saw is that the autopilot config is ignored

**Review by slevenick — COMMENTED** [review]
It looks like many tests are failing due to this error:
Error 403: Permission denied: Get resource manager tags: GCE_PERMISSION_DENIED: permission denied: for service-1234567@container-engine-robot.iam.gserviceaccount.com on TagKey

This might be tricky. I'm not sure why we are getting that error as it seems like the tags should be retrieved on the normal GET call?

**Comment by maxi-cit** [comment]
Hello @slevenick, I added the 'tagHoldAdmin' policy binding for the default GKE SA, is this happening in all or a few of the tests?

**Comment by slevenick** [comment]
> Hello @slevenick, I added the 'tagHoldAdmin' policy binding for the default GKE SA, is this happening in all or a few of the tests?

I think it's actually that you've added a tag key to a bunch of the existing tests? Can you isolate the addition of the tags to only a few tests rather than adding it to so many existing ones? It looks like this is causing the failure for at least `TestAccContainerCluster_resourceManagerTags`

The test `TestAccContainerCluster_withAutopilot` is failing with: `         Error: Reference to undeclared resource`

**Comment by maxi-cit** [comment]
Hello @slevenick, I just added 3 tests that contain the tag key. Could you point to the code you are referring to, please? I will verify the autopilot test. Regarding the diff issue. I will comment the auto pilot mode and see if that fixes it

**Comment by slevenick** [comment]
> Hello @slevenick, I just added 3 tests that contain the tag key. Could you point to the code you are referring to, please? I will verify the autopilot test. Regarding the diff issue. I will comment the auto pilot mode and see if that fixes it

I guess just the autopilot tests. I'm not sure why we need to add tags on the existing autopilot tests

**Comment by maxi-cit** [comment]
Hello folks, I isolated the autopilot test. And I will probably will open a PR separately. I am delaying because I ve getting this error while trying the tests locally. But I will send my fixes as I can verify the tests for clusters and node pools
A n1-standard-1 VM instance is currently unavailable in the us-central1-a zone. Alternatively, you can try your request again with a different VM hardware configuration or at a later time. For more information, see the troubleshooting documentation."

**Comment by slevenick** [comment]
Tests continue to fail with permission denied:
service-1234567@container-engine-robot.iam.gserviceaccount.com on TagKey "tagKeys/98765432" (or it may not exist).

**Comment by maxi-cit** [comment]
Hello @slevenick, I was able to reproduce this error. Indeed, it is a permission issue that is solved by granting the 'roles/resourcemanager.tagUser' to the default GKE Service Account (in this specific scenario). However, there seems to be a propagation issue in the current way iam_bindings works so cluster is created before its SA gets the permissions setup.

I am trying to re run the tests locally with a time_sleep resource. Will be back with more news soon!

**Comment by maxi-cit** [comment]
Hey @slevenick, could you run the tests please? I added some extra permissions on GKE and GCP SA. it should be fine now

**Comment by maxi-cit** [comment]
Hello @slevenick, could you give me a hint on why the replaying is failing, please?

File: mmv1/third_party/terraform/services/container/resource_container_node_pool_test.go.erb [hunk 17]
```diff
@@ -4208,3 +4263,336 @@ resource "google_container_node_pool" "without_confidential_boot_disk" {
...
+  }
+}
+
+resource "google_tags_tag_value" "value1" {
+  parent = "tagKeys/${google_tags_tag_key.key1.name}"
+  short_name = "foo1-%[2]s"
+  description = "For foo1 resources"
+}
+
+resource "google_tags_tag_key" "key2" {
```
[the remark this answers is not in the document]
Comment (maxi-cit) [reply mmv1/third_party/terraform/services/container/resource_container_node_pool_test.go.erb:4315]: added resource depency

File: mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb [hunk 18]
```diff
@@ -9443,3 +9533,189 @@ func testAccContainerCluster_withWorkloadALTSConfig(projectID, name, networkName
...
+    create = "30m"
+    update = "40m"
+  }
+
+  depends_on = [time_sleep.wait_120_seconds]
+}
+`, projectID, randomSuffix, clusterName, networkName, subnetworkName)
+}
+
+// TODO uncomment for auto_pilot_config
```
Comment (slevenick) [inline mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb:9618]: Can you remove this from this PR? It can be added in a later PR if needed

File: mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb [hunk 19]
```diff
@@ -9443,3 +9533,189 @@ func testAccContainerCluster_withWorkloadALTSConfig(projectID, name, networkName
...
+resource "google_project_iam_binding" "tagUser" {
+  project = "%[1]s"
+  role    = "roles/resourcemanager.tagUser"
+  members = [
+    "serviceAccount:service-${data.google_project.project.number}@container-engine-robot.iam.gserviceaccount.com",
+    "serviceAccount:${data.google_project.project.number}@cloudservices.gserviceaccount.com",
+  ]
+}
+
+resource "time_sleep" "wait_120_seconds" {
```
Comment (slevenick) [inline mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb:9560]: wait_300_seconds instead?

File: mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb [hunk 20]
```diff
@@ -9443,3 +9533,189 @@ func testAccContainerCluster_withWorkloadALTSConfig(projectID, name, networkName
...
+  deletion_protection = false
+  network    = "%[4]s"
+  subnetwork    = "%[5]s"
+
+  timeouts {
+    create = "30m"
+    update = "40m"
+  }
+
+  depends_on = [time_sleep.wait_120_seconds]
```
Comment (slevenick) [inline mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb:9613]: Why do we need this?

File: mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb [hunk 21]
```diff
@@ -2943,6 +2980,59 @@ func TestAccContainerCluster_withAutopilotNetworkTags(t *testing.T) {
 	})
 }
 
+// TODO uncomment for auto_pilot_config
```
Comment (slevenick) [inline mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb:2983]: Remove this until we need it

File: mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb [hunk 22]
```diff
@@ -5394,6 +5420,12 @@ func expandNodePoolAutoConfig(configured interface{}) *container.NodePoolAutoCon
 	if v, ok := config["network_tags"]; ok && len(v.([]interface{})) > 0 {
 		npac.NetworkTags = expandNodePoolAutoConfigNetworkTags(v)
 	}
+
+	// TODO uncomment for auto_pilot_config
```
Comment (slevenick) [inline mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb:5424]: Remove until we need it

File: mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb [hunk 23]
```diff
@@ -4142,6 +4149,25 @@ func resourceContainerClusterUpdate(d *schema.ResourceData, meta interface{}) er
 		log.Printf("[INFO] GKE cluster %s node pool auto config network tags have been updated", d.Id())
 	}
 
+	// TODO uncomment for auto_pilot_config
```
Comment (slevenick) [inline mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb:4152]: Remove until we need it

File: mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb [hunk 24]
```diff
@@ -1438,6 +1439,12 @@ func ResourceContainerCluster() *schema.Resource {
 								},
 							},
 						},
+						// TODO uncomment for auto_pilot_config
```
Comment (slevenick) [inline mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb:1442]: Remove until we need it

**Review by slevenick — CHANGES_REQUESTED** [review]

File: mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb (hunk 18 above)
Comment (maxi-cit) [reply mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb:9618]: removed

File: mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb (hunk 21 above)
Comment (maxi-cit) [reply mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb:2983]: removed

File: mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb (hunk 22 above)
Comment (maxi-cit) [reply mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb:5424]: removed

File: mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb (hunk 24 above)
Comment (maxi-cit) [reply mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb:1442]: removed

File: mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb (hunk 19 above)
Comment (maxi-cit) [reply mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb:9507]: Fixed, forget to set it back to 120s

**Review by slevenick — APPROVED** [review]
